src/sema/expression.rs

```diff
@@ -1061,6 +1061,20 @@ impl Expression {
                     Ok(Expression::Cast(*loc, to.clone(), Box::new(self.clone())))
                 }
             }
+            // Match any array with ArrayLength::AnyFixed if is it fixed for that dimension, and the
+            // element type and other dimensions also match
+            (Type::Array(from_elem, from_dim), Type::Array(to_elem, to_dim))
```

Comment:
Have you provided any tests for these new cast?
The cast function gave us headaches in the past.

Reply:
For now `ArrayLength::AnyFixed` is only used by `create_program_address()`. This covered by the `create_program_address()` tests.